Rubberduck.Core/UnitTesting/PermissiveObjectComparer.cs

```diff
@@ -16,6 +16,12 @@ public new bool Equals(object x, object y)
             var expected = x;
             var actual = y;
 
+            //VBA will coerce any numeric-apparent-value to test them
+            if (double.TryParse(expected.ToString(), out double numericExpected) && double.TryParse(actual.ToString(), out double numericActual))
```

Comment:
The approach to compare as doubles first is a bit naive, but I think doing it the correct way can wait until we implement the equality evaluation for the expression evaluation engine to spec. (As an example, when VBA compares a double with a single, they are compared as single.)